runtime/drivers/sql_store.go

```diff
@@ -0,0 +1,33 @@
+package drivers
+
+import (
+	"context"
+)
+
+type Field struct {
+	Name string
+	Type string
+}
+
+type Schema []Field
+
+// SQLStore is implemented by drivers capable of running sql queries and generating an iterator to consume results.
+// In future the results can be produced in other formats like arrow as well.
+// May be call it DataWarehouse to differentiate from OLAP or postgres?
+type SQLStore interface {
+	Exec(ctx context.Context, src *DatabaseSource) (RowIterator, error)
```

Comment:
1. Call it `Query` ("exec" is usually for statements that don't return values. On `OLAPStore`, the `Execute` function should be refactored to `Query`, just never got it done...)

2. It feels weird that this takes a `DatabaseSource` parameter since that is related to the transporter interface. It would be better to pass the relevant parameters directly or have an options type dedicated to this function (like how `OLAPStore` has `drivers.Statement`).